test/src/unit-vfs.cc

```diff
@@ -202,6 +202,76 @@ TEST_CASE("VFS: Test long local paths", "[vfs][long-paths]") {
   }
 }
 
+TEST_CASE("VFS: copy_file", "[vfs][copy_file]") {
+  LocalFsTest src_fs({0}), dst_fs({0});
+  URI src_path = src_fs.temp_dir_.add_trailing_slash();
+  URI dst_path = dst_fs.temp_dir_.add_trailing_slash();
+
+  ThreadPool compute_tp(4);
+  ThreadPool io_tp(4);
+  Config config = set_config_params();
+  VFS vfs{
+      &g_helper_stats, g_helper_logger().get(), &compute_tp, &io_tp, config};
+
+  size_t test_str_size = 0;
+  SECTION("Filesize = 0 MB") {
+    test_str_size = 0;
+  }
+  SECTION("Filesize = 1 MB") {
+    test_str_size = 1048576;
+  }
+  SECTION("Filesize = 10 MB") {
+    test_str_size = 10 * 1048576;
+  }
+  SECTION("Filesize = 100 MB") {
+    test_str_size = 100 * 1048576;
+  }
+  SECTION("Filesize = 150 MB") {
+    test_str_size = 150 * 1048576;
+  }
+  std::string test_str(test_str_size, 'a');
+  REQUIRE(test_str.size() == test_str_size);
+
+  // Create src_file and write data to it.
+  auto src_file = URI(src_path.to_string() + "src_file");
+  REQUIRE_NOTHROW(vfs.touch(src_file));
+  test_str_size = test_str.size();
+  REQUIRE_NOTHROW(vfs.write(src_file, test_str.data(), test_str_size));
+  require_tiledb_ok(vfs.close_file(src_file));
+
+  // copy_file src -> dst using chunked-buffer I/O.
+  // Note: it doesn't matter if the dst file exists; copy will create on write.
+  auto dst_file = URI(dst_path.to_string() + "dst_file");
+  REQUIRE_NOTHROW(vfs.chunked_buffer_io(src_file, dst_file));
```

Comment:
This looks odd as the top-level API, shouldn't this just be `copy`?
Users just want fast copy, they don't care about the implementation details

Reply:
See https://github.com/TileDB-Inc/TileDB/pull/5636#discussion_r2350284551. Calling `copy` between URIs of the same file system would bypass the logic added in this PR. We directly call `chunked_file_io` to test it, without having to spin up a cloud storage emulator.